### PR TITLE
[clkmgr] Release measure lock on clock calibration loss

### PR DESCRIPTION
**Pull request by tjaychen**
- addresses https://github.com/lowRISC/opentitan/issues/12523

The basic problem described in the issue is measurement controls survive
low power entry/exit.  However, the clock calibration settings do not
(the ast registers are reset).

This means upon low power exit, assuming measurements were configured
prior, there is an immediate mismatch between count and clock.

To address this, the clkmgr now observes when the clocks lose its
calibrated state by observing `ast_init_done`.  When this is seen,
clkmgr automatically disables measurements and allows software to
reconfigure.

While this solves the existing issue, it does mean that if an attacker
were able to fake an unlock, the measure controls would turn off.
However, there is no good way to handle that situation in this boundary
since that attack can look no different from a low power entry.

It will be up to the system to place defenses elsewhere to check that a
calibration lost is intended behavior.

### CODE REVIEW

**Comment by tjaychen** [comment]
@Jacob-Levy 
@zi-v 
is `ast_init_done` an accurate signal to use to indicate clocks are no longer calibrated? Do we want something with finer granularity?

**Comment by zi-v** [comment]
I think that ast_init_done is a good indication.
You may want to consider only TL-UL reset for clock measurement deactivation (slightly more secured), but I don't think it is a must because deactivating the clock measurement is far enough from total breach,

File: sw/device/tests/BUILD [hunk 1]
```diff
@@ -1062,3 +1062,18 @@ opentitan_functest(
         "//sw/device/lib/testing/test_framework:ottf_main",
     ],
 )
+
```
Comment (jdonjdon) [inline sw/device/tests/BUILD:1065]: please update this file to fix build error
    deps = [
        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
        "//sw/device/lib/base:memory",
        "//sw/device/lib/base:mmio",
        "//sw/device/lib/dif:sensor_ctrl",
        "//sw/device/lib/runtime:log",
        "//sw/device/lib/testing:aon_timer_testutils",
        "//sw/device/lib/testing:clkmgr_testutils",
        "//sw/device/lib/testing:pwrmgr_testutils",
        "//sw/device/lib/testing/test_framework:check",
        "//sw/device/lib/testing/test_framework:ottf_main",
    ],

**Review by jdonjdon — CHANGES_REQUESTED** [review]

File: sw/device/tests/BUILD (hunk 1 above)
Comment (tjaychen) [reply sw/device/tests/BUILD:1065]: thanks @jdonjdon! do you want me to merge the test of this PR? or do you want to do it separately?
Both are okay with me.

**Comment by tjaychen** [comment]
> I think that ast_init_done is a good indication. You may want to consider only TL-UL reset for clock measurement deactivation (slightly more secured), but I don't think it is a must because deactivating the clock measurement is far enough from total breach,

it's funny you mention that, because originally that was the path i was going down. But then I thought...why not use a signal whose polarity will always be a certain state when TLUL resets? That's what led me to ast_init_done.

File: sw/device/tests/BUILD (hunk 1 above)
Comment (jdonjdon) [reply sw/device/tests/BUILD:1065]: please go ahead merge.
I don't have merge permit yet ;)

**Review by eunchan — APPROVED** [review]
`ast_init_done` is registered output. Looks good to me.

**Pull request merged**